inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/DataProviderResolver.java

```diff
@@ -0,0 +1,81 @@
+package rocks.inspectit.oce.core.instrumentation.config;
+
+import lombok.val;
+import org.springframework.stereotype.Component;
+import rocks.inspectit.oce.core.config.model.instrumentation.InstrumentationSettings;
+import rocks.inspectit.oce.core.config.model.instrumentation.dataproviders.GenericDataProviderSettings;
+import rocks.inspectit.oce.core.instrumentation.config.model.InstrumentationConfiguration;
+import rocks.inspectit.oce.core.instrumentation.config.model.ResolvedGenericDataProviderConfig;
+import rocks.inspectit.oce.core.instrumentation.dataprovider.generic.DataProviderGenerator;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class DataProviderResolver {
+
+    /**
+     * Resolves {@link InstrumentationConfiguration#getDataProviders()}.
+     * Returns a map mapping the names of providers to their resolved configurations
+     *
+     * @param builder the builder to resolve
+     * @param source  the input configuration
+     * @return the updated builder
+     */
+    Map<String, ResolvedGenericDataProviderConfig> resolveProviders(InstrumentationSettings source) {
+        Map<String, ResolvedGenericDataProviderConfig> resultMap = new HashMap<>();
+        for (val entry : source.getDataProviders().entrySet()) {
+            String name = entry.getKey();
+            val conf = entry.getValue();
+
+            HashMap<String, String> additionalInputs = new HashMap<>(conf.getInput());
+
+            val result = ResolvedGenericDataProviderConfig.builder()
+                    .name(name)
+                    .importedPackages(conf.getImports());
+
+            resolveSpecialVariables(additionalInputs, result);
+            resolveArgumentVariables(additionalInputs, result);
+            resolveAdditionalInputs(additionalInputs, result);
+            resolveBody(conf, result);
+
+            resultMap.put(name, result.build());
+        }
+        return resultMap;
+    }
+
+    private void resolveBody(GenericDataProviderSettings conf, ResolvedGenericDataProviderConfig.ResolvedGenericDataProviderConfigBuilder result) {
+        if (conf.getValue() != null) {
+            result.valueBody("return " + conf.getValue() + ";");
+        } else {
+            result.valueBody(conf.getValueBody());
+        }
+    }
+
+    private void resolveAdditionalInputs(HashMap<String, String> additionalInputs, ResolvedGenericDataProviderConfig.ResolvedGenericDataProviderConfigBuilder result) {
+        //everything that has not been removed is an additional argument
+        for (val additionalVarEntry : additionalInputs.entrySet()) {
```

Comment:
`additionalInputs.entrySet().forEach(result::additionalArgumentType)` a bit more of Java8 style :-D

Reply:
This one and the comment above are left-overs from where I had to write to a local variable outside of the loop :-D